src/Integration.Vsix/ErrorList/SonarErrorListDataSource.cs

```diff
@@ -212,10 +212,18 @@ public void Refresh(IEnumerable<string> affectedFilePaths)
                 foreach (var factory in factories)
                 {
                     var snapshot = factory.CurrentSnapshot;
-                    
-                    if (snapshot.FilesInSnapshot.Any(snapshotPath => affectedFilePaths.Any(affected => PathHelper.IsMatchingPath(snapshotPath, affected))))
+                    var isSnapshotAffected = snapshot.FilesInSnapshot.Any(snapshotPath => affectedFilePaths.Any(affected => PathHelper.IsMatchingPath(snapshotPath, affected)));
+
+                    if (isSnapshotAffected)
                     {
-                        snapshot.IncrementVersion();
+                        if (snapshot.HasNonNavigableIssues)
+                        {
+                            factory.UpdateSnapshot(factory.CurrentSnapshot.CreateUpdatedSnapshot());
+                        }
+                        else
+                        {
+                            snapshot.IncrementVersion();
+                        }
```

Comment:
Add comment below about lines 230+ re clearing selection based on previous contents
Rename snapshot to oldSnapshot